wgsl/index.bs

```diff
@@ -11985,6 +11985,38 @@ Note: The result is not mathematically meaningful when `abs(e)` &ge; 1.
     <td>Returns the dot product of `e1` and `e2`.
 </table>
 
+### `dot4U8Packed` ### {#dot4U8Packed-builtin}
+<table class='data builtin'>
+  <tr algorithm="dot4U8Packed">
+    <td style="width:10%">Overload
+    <td class="nowrap"><xmp highlight=rust>
+@const fn dot4U8Packed(e1: u32,
+                       e2: u32) -> u32
+</xmp>
+  <tr>
+    <td>Description
+    <td> `e1` and `e2` are interpreted as vectors with four 8-bit unsigned integer components.
+    Vector components follow byte significance order with the lowest-numbered component
+    stored in the least significant byte.
+    Return the unsigned integer dot product of these two vectors.
+</table>
+
+### `dot4I8Packed` ### {#dot4I8Packed-builtin}
+<table class='data builtin'>
+  <tr algorithm="dot4I8Packed">
+    <td style="width:10%">Overload
+    <td class="nowrap"><xmp highlight=rust>
+@const fn dot4I8Packed(e1: u32,
+                       e2: u32) -> I32
+</xmp>
+  <tr>
+    <td>Description
+    <td> `e1` and `e2` are interpreted as vectors with four 8-bit signed integer components.
+    Vector components follow byte significance order with the lowest-numbered component
+    stored in the least significant byte.
+    Return the signed integer dot product of these two vectors.
```

Comment:
I think we have to say that each component is sign-extended to i32 before performing the multiply, and then the add operations are done in WGSL i32 with wrapping behaviour.

For example, consider components  0xff as bits, which is -1, and 0x80 as bits which is -128.
Then -1 * -128 is 128.
But when we multiply as i8 integers we have:
```
    0xff
 *  0x80
= 0x3f80
```
Which is not 128.